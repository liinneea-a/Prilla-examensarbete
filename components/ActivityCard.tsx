import { Text } from './Themed';
import { Image, ImageBackground, StyleSheet, View } from 'react-native';
import { Review, User } from '../Interfaces';
import { ReviewCard } from './ReviewCard';
import { useEffect, useState } from 'react';
import { getOneDocById } from '../helper';

interface Props {
	review: Review;
}

export const ActivityCard = ({ review }: Props) => {
	const [author, setAuthor] = useState<User>();

	useEffect(() => {
		getReviewAuthor();
	}, []);

	const getReviewAuthor = async () => {
		let data = await getOneDocById('users', review.userID);
		if (data) {
			setAuthor(data as User);
		}
	};

	return (
		<View>
			<ImageBackground
				source={{uri: review.photo}}
				resizeMode="cover"
				style={styles.image}
			>
				<View style={styles.userInfo}>
					<Text>User Pic</Text>
					<Text style={styles.username}>{author?.displayName}</Text>
				</View>
				<ReviewCard key={review.id} review={review} />
			</ImageBackground>
		</View>
	);
};
const styles = StyleSheet.create({
	image: {
		marginTop: 10,
		marginBottom: 10,
		paddingTop: 10,
		paddingBottom: 10,
		minHeight: 230,
		justifyContent: 'space-between',
		alignItems: 'center',
	},
	userInfo: {
		flexDirection: 'row',
		width: '100%',
	},
	username: {
		fontWeight: 'bold',
		marginLeft: 10,
	},
});
