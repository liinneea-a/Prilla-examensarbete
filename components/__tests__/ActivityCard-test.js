import * as React from "react";
import renderer from "react-test-renderer";
import { string } from "yup";
import { ActivityCard } from "../ActivityCard";

const review = {
  photo: string,
  id: string,
};

it("should contain a placeholder and author name be to undefined", () => {
  const card = renderer.create(<ActivityCard />);
  const instance = card.root;
  const placeholder = instance.findByProps({ testID: "placeholder" });
  const author = card.root.findByProps({ testID: "author" });
  expect(placeholder.props.children).toMatch("User Pic");
  expect(author.props.children).toBeUndefined();
});
