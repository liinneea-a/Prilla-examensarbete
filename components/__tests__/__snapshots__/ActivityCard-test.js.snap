// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain to text tags 1`] = `
<View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginTop": 10,
        "minHeight": 230,
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <Image
      resizeMode="cover"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="image"
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
            undefined,
          ]
        }
        testID="placeholder"
      >
        User Pic
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
            Object {
              "fontWeight": "bold",
              "marginLeft": 10,
            },
          ]
        }
        testID="author"
      />
    </View>
  </View>
</View>
`;
